src/twiliosim.cr

```diff
@@ -31,85 +32,136 @@ class Twiliosim::Server
       response.to_json(context.response)
     when %r(/Accounts/(.+)/Calls.*)
       account_sid = $1
-      body = request.body
-      unless body
-        plain_response(context, 400, "Request body is missing")
-        return
-      end
-      body = body.gets_to_end
-      unless body
-        plain_response(context, 400, "Request body is missing")
-        return
-      end
-      if body.blank?
-        plain_response(context, 400, "Request body is missing")
-        return
-      end
-      params = HTTP::Params.parse(body)
-      unless params.has_key?("Url")
-        plain_response(context, 400, "Body param 'Url' is missing")
-        return
-      end
-      verboice_url = params["Url"]
-      unless params.has_key?("From")
-        plain_response(context, 400, "Body param 'From' is missing")
+      handle_call_request(context, account_sid)
+    else
+      plain_response(context, 404, "404 Not Found")
+    end
+  end
+
+  private def handle_incoming_phone_numbers_request(context)
+    context.response.status_code = 200
+    context.response.content_type = "application/json"
+    response = {"sid" => UUID.random().to_s()}
+    response.to_json(context.response)
+  end
+
+  private def handle_call_request(context, account_sid)
+    body_params = get_body_params(context.request.body)
+
+    if (body_params["error"])
+      plain_response(context, 400, body_params["error"])
+      return
+    end
+
+    from = body_params["from"]
+    to = body_params["to"]
+    verboice_url = body_params["verboice_url"]
+
+    unless from && to && verboice_url
+      plain_response(context, 500, "Internal error getting the request params")
+      return
+    end
+
+    call = start_call(to, from, account_sid)
+    response_call_created(context, call.sid)
+    spawn do
+      sleep 1.seconds # give Verboice a sec just to process the response
+      handle_created_call(verboice_url.not_nil!, call)
```

Comment:
@bcardiff I wasn't able to avoid the `not_nil!` call here.

Do you know why when I remove it I get this error?

![error](https://user-images.githubusercontent.com/39921597/98534411-33bc0980-2263-11eb-83b7-732ba75fd7d5.png)